include/yaml-cpp/node/convert.h

```diff
@@ -161,83 +167,272 @@ struct convert<bool> {
   YAML_CPP_API static bool decode(const Node& node, bool& rhs);
 };
 
-// std::map
-template <typename K, typename V>
-struct convert<std::map<K, V>> {
-  static Node encode(const std::map<K, V>& rhs) {
+template <class T> class node_push_backer
+{
+ private:
+  Node* m_node;
+
+ public:
+  typedef void value_type;
+  typedef void difference_type;
+  typedef void pointer;
+  typedef void reference;
+  typedef std::output_iterator_tag iterator_category;
+
+  explicit node_push_backer(Node& node) : m_node(&node) {}
+
+  node_push_backer& operator=(const T& t)
+  {
+      m_node->push_back(t);
+      return *this;
+  }
+
+  node_push_backer& operator*()     { return *this; }
+  node_push_backer& operator++()    { return *this; }
+  node_push_backer  operator++(int) { return *this; }
+};
+
+template <class Container> class push_back_iterator {
+ private:
+  Container* m_c;
+
+ public:
+  typedef void value_type;
+  typedef void difference_type;
+  typedef void pointer;
+  typedef void reference;
+  typedef std::output_iterator_tag iterator_category;
+
+  push_back_iterator(Container& container) : m_c(&container) {}
+
+  push_back_iterator& operator=(const Node& node)
+  {
+#if defined(__GNUC__) && __GNUC__ < 4
+      // workaround for GCC 3:
+      m_c->push_back(node.template as<typename Container::value_type>());
+#else
+      m_c->push_back(node.as<typename Container::value_type>());
+#endif
+      return *this;
+  }
+
+  push_back_iterator& operator*()     { return *this; }
+  push_back_iterator& operator++()    { return *this; }
+  push_back_iterator  operator++(int) { return *this; }
+};
+
+template <class Container> 
+push_back_iterator<Container> push_backer(Container& container)
+{
+  return push_back_iterator<Container>(container);
+}
+
+template <
+  class Key, 
+  class T, 
+  class ContainerValueType = std::pair<const Key, T>
+> class node_inserter
+{
+ private:
+  Node* m_node;
+
+ public:
+  typedef void value_type;
+  typedef void difference_type;
+  typedef void pointer;
+  typedef void reference;
+  typedef std::output_iterator_tag iterator_category;
+
+  explicit node_inserter(Node& node) : m_node(&node) {}
+
+  node_inserter& operator=(const ContainerValueType& t)
+  {
+      m_node->force_insert(t.first, t.second);
+      return *this;
+  }
+
+  node_inserter& operator*()     { return *this; }
+  node_inserter& operator++()    { return *this; }
+  node_inserter  operator++(int) { return *this; }
+};
+
+template <class T> struct convert_as_sequence
+{
+  static Node encode(const T& sequence)
+  {
+    Node node(NodeType::Sequence);
+    
+    std::copy(sequence.begin(), sequence.end(), 
+      node_push_backer<typename T::value_type>(node));
+
+    return node;
+  }
+  static bool decode(const Node& node, T& sequence)
+  {
+    if (!node.IsSequence())
```

Comment:
braces around single-line "if"

(throughout, including code that already exists)